Update router_state.ts

Wilt opened the pull request:
It should be possible to explicitly type route data. 
This PR allows for optional typing of ActivatedRoute and ActivatedRouteSnapshot.
Fallback to the original "Data" type for backwards compatibility.

Wilt commented:
@mmalerba Thanks for merging my other two suggestions https://github.com/angular/components/pull/30324 and https://github.com/angular/components/pull/30477 for optional typing.
This one is in the same line, but would have an even greater impact for all of those who ❤ strict typing (like @msmallest).

mmalerba commented:
It looks like this PR is to your own fork, I assume you meant to send it to angular/angular?

Wilt commented:
Yes, I guess I did that wrong. 
I will need to resolve that at a later time. I will notify you when done. 
Thank you for your patience.

Wilt committed:
feat(router): optional typing of route data

Add optional typing to ActivatedRoute and ActivatedRouteSnapshot data to enforce strict route data typing.

Closes #30477

Wilt commented:
@mmalerba 

See here for a PR against the correct repository: https://github.com/angular/angular/pull/60064

Thanks again!
